src/pkg/packager/actions/actions.go

```diff
@@ -89,82 +90,55 @@ func runAction(ctx context.Context, defaultCfg v1alpha1.ZarfComponentActionDefau
 		spinner.Errorf(err, "Error mutating command: %s", cmdEscaped)
 	}
 
-	duration := time.Duration(actionDefaults.MaxTotalSeconds) * time.Second
-	timeout := time.After(duration)
+	// Keep trying until the max retries or timeout is reached.
+	actionCtx := ctx
+	if actionDefaults.MaxTotalSeconds > 0 {
+		var actionCancel context.CancelFunc
+		actionCtx, actionCancel = context.WithTimeout(ctx, time.Duration(actionDefaults.MaxTotalSeconds)*time.Second)
+		defer actionCancel()
+	}
+	err = retry.Do(func() error {
+		if out, err = actionRun(actionCtx, actionDefaults, cmd, actionDefaults.Shell, spinner); err != nil {
+			return err
+		}
+		out = strings.TrimSpace(out)
 
-	// Keep trying until the max retries is reached.
-retryCmd:
-	for remaining := actionDefaults.MaxRetries + 1; remaining > 0; remaining-- {
-		// Perform the action run.
-		tryCmd := func(ctx context.Context) error {
-			// Try running the command and continue the retry loop if it fails.
-			if out, err = actionRun(ctx, actionDefaults, cmd, actionDefaults.Shell, spinner); err != nil {
+		// If an output variable is defined, set it.
+		for _, v := range action.SetVariables {
```

Comment:
Should this be retried? It it seems like this was done because implementing it would have been difficult in the old code. It seems a lot smarter to only retry the action on errors and not the setting on variables as I assume the output variables will be the same every time the actions run.